javascript/ql/lib/semmle/javascript/frameworks/Restify.qll

```diff
@@ -19,23 +19,24 @@ module Restify {
   /**
    * A Restify route handler.
    */
-  class RouteHandler extends Http::Servers::StandardRouteHandler, DataFlow::ValueNode {
-    Function function;
-
-    RouteHandler() {
-      function = astNode and
-      any(RouteSetup setup).getARouteHandler() = this
-    }
-
+  abstract class RouteHandler extends Http::Servers::StandardRouteHandler, DataFlow::ValueNode {
     /**
      * Gets the parameter of the route handler that contains the request object.
      */
-    Parameter getRequestParameter() { result = function.getParameter(0) }
+    Parameter getRequestParameter() { result = this.asExpr().(Function).getParameter(0) }
 
     /**
      * Gets the parameter of the route handler that contains the response object.
      */
-    Parameter getResponseParameter() { result = function.getParameter(1) }
+    Parameter getResponseParameter() { result = this.asExpr().(Function).getParameter(1) }
+  }
+
+  /**
+   * A standard Restify route handler.
+   */
+  abstract class StandardRouteHandler extends Http::Servers::StandardRouteHandler,
```

Comment:
Thanks! Just out of curiosity, why is it necessary to split the routeHandler into these two?